src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+
+namespace System.Device.Pwm
+{
+    /// <summary>
+    /// Represents a PWM channel running on Unix.
+    /// </summary>
+    internal class UnixPwmChannel : PwmChannel
+    {
+        private readonly int _chip;
+        private readonly int _channel;
+        private double _frequency;
+        private double _dutyCyclePercentage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnixPwmChannel"/> class.
+        /// </summary>
+        /// <param name="chip">The PWM chip number.</param>
+        /// <param name="channel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity of the PWM signal.</param>
+        public UnixPwmChannel(
+            int chip,
+            int channel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh)
+        {
+            _chip = chip;
+            _channel = channel;
+            Validate();
+            Open();
+            Thread.Sleep(100);  // TODO: Need a better solution to delay for available file.
+
+            SetFrequency(frequency);
+            DutyCyclePercentage = dutyCyclePercentage;
+
+            // NOTE: There appears to be a lower-level bug that won't allow the
+            // polarity to be changed until a valid period has been written.
```

Comment:
When you say period, I assume you mean DutyCycle? In that case, should Validate also ensure that frequency and dutycycle are valid?

Reply:
No.  When the RPi boots and you open/export the pwm channel (at least for the first time), the period is set to '0'.  For some reason you must change the period (of frequency) to a valid value before changing the polarity.  It should be okay since the period is set to a valid value in the SetFrequency(...) method a few lines above and is the only place polarity should be changed.  I noticed others were seeing this issue in other IoT libraries while researching this.